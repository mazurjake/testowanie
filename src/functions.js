function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

export const getAsyncData = async ()=>{
    try{
        await sleep(2000)
        return ['super data']
    }
   catch {
        return ['failed']
   }
}
