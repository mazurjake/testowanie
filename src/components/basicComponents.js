import React, { useState, useEffect } from 'react';
import { getAsyncData } from "../functions";

export const BasicFunctionalComponent = () =><div>QARBON RULES</div>

export const PropsRenderingTest = ({name, color}) =><div className={color}>{name}</div>

export const AsyncDataComponent = ({func, condition ,content }) => {
    const [data, setData] = useState(null)

    useEffect(()=>{
        if(condition) func()
        else {
            const dane = getAsyncData()
            setData(dane)
        }
    },[])

    return(
    <div>
        {data ? data.map((item, index)=>{return <div  key={index}> {item} {content}</div>}): null}
    </div>)
}

export const TriggerEffect = ({func}) => {
    const [condition, setCondition] = useState(false)

    useEffect(()=>{
        func()
    },[condition])

    return(<div><button onClick={()=>{setCondition(!condition)}}></button></div>)
}

export const getMajonez = (callback) =>{
    return callback()
}

