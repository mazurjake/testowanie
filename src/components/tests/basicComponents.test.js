import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({ adapter: new Adapter() });
import 'regenerator-runtime/runtime';
import Enzyme,{shallow,  mount} from "enzyme";
import React from 'react'
import { BasicFunctionalComponent, PropsRenderingTest, AsyncDataComponent, TriggerEffect} from "../basicComponents";
import * as module from '../../functions'


describe('For BasicFunctionalComponent', ()=>{
    test('Should render correct snapshot',()=>{
        const wrapper = shallow(<BasicFunctionalComponent/>)
        expect(wrapper.html()).toMatchSnapshot()
    })
});
describe('For PropsRenderingTest', ()=>{
    test('Should render correct snapshot (1)',()=>{
        const wrapper = shallow(<PropsRenderingTest />)
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('Should render correct snapshot (2)',()=>{
        const default_props = {
            name: 'Maciej',
            color: 'Red',
        }
        const wrapper = shallow(<PropsRenderingTest  {...default_props}/>)
        expect(wrapper.html()).toMatchSnapshot()
    })
});
describe('For AsyncDataComponent', ()=>{
    test('Should render correct snapshot',()=>{
        const default_props ={func: jest.fn()}
        jest.spyOn(module, 'getAsyncData').mockReturnValue(['replaced the data'])
        const wrapper = mount(<AsyncDataComponent {...default_props}/>)
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('Should call props.func',()=>{
        const default_props ={func: jest.fn(), condition:true}
        mount(<AsyncDataComponent {...default_props}/>)
        expect(default_props.func).toHaveBeenCalled()
    })
    test('Should call getAsyncData',()=>{
        const default_props ={func: jest.fn()}
        jest.spyOn(module, 'getAsyncData').mockReturnValue(['replaced the data'])
        mount(<AsyncDataComponent {...default_props}/>)
        expect(module.getAsyncData).toHaveBeenCalled()
    })
});
describe('For TriggerEffect', ()=>{
    test('Should render correct snapshot',()=>{
        const default_props ={func: jest.fn()}
        const wrapper = mount(<TriggerEffect {...default_props}/>)
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('Should call props.func once',()=>{
        const default_props ={func: jest.fn()}
        mount(<TriggerEffect {...default_props}/>)
        expect(default_props.func).toHaveBeenCalledTimes(1)
    })
    test('Should call props.func twice',()=>{
        const default_props ={func: jest.fn()}
        const wrapper = mount(<TriggerEffect {...default_props}/>)
        wrapper.find('button').simulate('click')
        expect(default_props.func).toHaveBeenCalledTimes(2)
    })
});


// "setupFilesAfterEnv": [
//     "<rootDir>/src/setupTests.js"
// ],
