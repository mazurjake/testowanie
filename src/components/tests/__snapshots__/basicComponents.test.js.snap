// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`For AsyncDataComponent Should render correct snapshot 1`] = `"<div><div> replaced the data</div></div>"`;

exports[`For BasicFunctionalComponent Should render correct snapshot 1`] = `"<div>hey dude</div>"`;

exports[`For PropsRenderingTest Should render correct snapshot (1) 1`] = `"<div></div>"`;

exports[`For PropsRenderingTest Should render correct snapshot (2) 1`] = `"<div class=\\"Red\\">Maciej</div>"`;

exports[`For TriggerEffect Should render correct snapshot 1`] = `"<div><button></button></div>"`;
